addons/web/controllers/main.py

```diff
@@ -1991,7 +1995,11 @@ def content_type(self):
         return 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
 
     def filename(self, base):
-        return base + '.xlsx'
+        if base not in request.env:
+            return base + '.xlsx'
+
+        model_desc = request.env['ir.model']._get(base).name
+        return '%s (%s).xlsx' % (model_desc or '', base)
```

Comment:
If the pivot export is translatable, maybe this should be as well in case there are cultures which need to invert the two terms.